engine/src/Player.cpp

```diff
@@ -18,18 +18,21 @@ void Player::setSpeed(float speed) {
     }
 }
 
-const Character *Player::getTarget() const {
+
+const std::weak_ptr<Entity> &Player::getTarget() const {
```

Comment:
You should not return references with smart pointers because you may distort the reference counting.